### ISSUE TITLE
get_roles inconsistent 

### ISSUE THREAD

**Issue by pbauer** [plone/plone.api#354]
For users: `get_roles('foo', obj=bar, inherit=False)` uses `obj.get_local_roles_for_userid(username)` which ignores LocalRoles-Adapters.

For groups it uses `lrmanager.getRolesInContext(plone_user, obj)` which includes local_roles set on parents.

Both are wrong should instead return roles that are set only on this object including those provided by LocalRoles-Adapters or from groups that the user is a member of but ignore global roles and roles from parents.

I'm working on a pull-request.

**Comment by pbauer** [comment]
This solution in https://github.com/plone/plone.api/pull/355 is not really beautiful but seems to cover all the use-cases:

```python
plone_user = user.getUser()
principal_ids = list(plone_user.getGroups())
principal_ids.insert(0, plone_user.getId())
roles = set([])
pas = portal.get_tool('acl_users')
for _, lrmanager in pas.plugins.listPlugins(ILocalRolesPlugin):
    for adapter in lrmanager._getAdapters(obj):
        for pid in principal_ids:
            roles.update(adapter.getRoles(pid))
return list(roles)
```

It includes basic local_roles (`obj.__ac_local_roles__`) for the user and all groups that he is a member of. It also handles LocalRoles-Adapters. For groups the only difference is the first line: `plone_user = super(group.__class__, group)`.

**Comment by thet** [comment]
Fixed by #355.
